### DISCUSSION TITLE
"Callback" in Event subscription

### DISCUSSION THREAD

**Discussion by danmoser** [FreeOpcUa/opcua-asyncio#997] [Q&A] [answered]
Hi there,
I'm getting into Events now and I'm wondering if it's possible to trigger an event every time a client subscribe and unsubscribe from a given object. Example:

```
sub = await client.create_subscription(0, handler)
handle = await sub.subscribe_events(obj) 
# an event is triggered on the server after subscription
await sub.unsubscribe(handle)
# another event is triggered on the server before unsubscription
```
Thanks in advance!

**Comment by schroeder-** [comment, marked as answer]
I don't think there is anything currently. Maybe there is a way to monkey patch something or read the list of subscriped variables in a loop.

**Reply by danmoser** [reply]
Thanks @schroeder- . Could you point me to an example to get the subscribed variables in the server?

**Reply by schroeder-** [reply]
```Python
for handle, data in server.iserver.aspace._handle_to_attribute_map:
    nid, attrb = data
```
